File: app-mobile/trade_it/lib/models/sub_category.dart
class SubCategory {
  int? subcategoryId;
  String? name;
  var category;
  int? categoryId;
  List? products;

  SubCategory(
      {this.subcategoryId,
      this.name,
      this.category,
      this.categoryId,
      this.products});

  SubCategory.fromJson(Map<String, dynamic> json) {
    subcategoryId = json['subcategoryId'];
    name = json['name'];
    category = json['category'];
    categoryId = json['categoryId'];
    products = json['products'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['subcategoryId'] = this.subcategoryId;
    data['name'] = this.name;
    data['category'] = this.category;
    data['categoryId'] = this.categoryId;
    data['products'] = this.products;
    return data;
  }
}

File: Backend (ASP .Net Web Api)/API/DatabaseLayer/Models/Subcategory.cs
namespace DatabaseLayer.Models
{
    public class Subcategory
    {
        public int SubcategoryId { get; set; }
        public string Name { get; set; }


        public virtual Category Category { get; set; }
        public int CategoryId { get; set; }

        public virtual ICollection<Product> Products { get; set; }
    }
}
